WebScreenshotCreator exception

Some tests not using WebSteps or driver at all can fail on after story screenshot.
We should investigate and if possible add some condition to trigger only for running browser.

Example stack trace:
```net.homecredit.cmsgov.testautomation.services.DataCheckService$IsEmptyException: Expected data not found: 
select *
from cmsf.iso_message_maintenance imm
where imm.payment_instrument_id = :code
order by imm.id desc
	at net.homecredit.cmsgov.testautomation.services.DataCheckService.isNotEmpty(DataCheckService.java:110)
	at net.homecredit.cmsgov.testautomation.services.DataCheckService.checkExpectedDataListAgainstDataFromDatabase(DataCheckService.java:86)
	at net.homecredit.cmsgov.testautomation.services.DataCheckService.checkExpectedDataListAgainstDataFromDatabase(DataCheckService.java:48)
	at net.homecredit.cmsgov.testautomation.steps.DbSteps.resultEqualsQueryExecution(DbSteps.java:29)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.jbehave.core.steps.StepCreator$ParametrisedStep.perform(StepCreator.java:822)
	at org.jbehave.core.embedder.PerformableTree$FineSoFar.run(PerformableTree.java:353)
	at org.jbehave.core.embedder.PerformableTree$PerformableSteps.perform(PerformableTree.java:1237)
	at org.jbehave.core.embedder.PerformableTree$AbstractPerformableScenario.perform(PerformableTree.java:1108)
	at org.jbehave.core.embedder.PerformableTree$AbstractPerformableScenario.performRestartableSteps(PerformableTree.java:1087)
	at org.jbehave.core.embedder.PerformableTree$AbstractPerformableScenario.perform(PerformableTree.java:1099)
	at org.jbehave.core.embedder.PerformableTree$NormalPerformableScenario.perform(PerformableTree.java:1134)
	at org.jbehave.core.embedder.PerformableTree$PerformableScenario.perform(PerformableTree.java:1022)
	at org.jbehave.core.embedder.PerformableTree$PerformableStory.performScenarios(PerformableTree.java:909)
	at org.jbehave.core.embedder.PerformableTree$PerformableStory.perform(PerformableTree.java:877)
	at org.jbehave.core.embedder.PerformableTree.performCancellable(PerformableTree.java:432)
	at org.jbehave.core.embedder.PerformableTree.perform(PerformableTree.java:399)
	at org.jbehave.core.embedder.StoryManager$EnqueuedStory.call(StoryManager.java:293)
	at org.jbehave.core.embedder.StoryManager$EnqueuedStory.call(StoryManager.java:266)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
13:12:36.870 [pool-11-thread-1] INFO  o.j.c.i.web.WebScreenshotCreator - Taking FAILED screenshot will place it in ./target/reports
13:12:36.871 [pool-11-thread-1] ERROR o.j.runner.reporter.LoggingReporter - Failed step: org.jbehavesupport.core.web.WebSteps.afterFailedScenario() cause: {}
org.jbehave.core.failures.BeforeOrAfterFailed: Method afterFailedScenario (annotated with @AfterScenario in class org.jbehavesupport.core.web.WebSteps) failed: java.io.UncheckedIOException: java.nio.file.NoSuchFileException: ./target/reports
	at org.jbehave.core.steps.StepCreator$BeforeOrAfterStep.perform(StepCreator.java:707)
	at org.jbehave.core.steps.StepCreator$DelegatingStep.perform(StepCreator.java:668)
	at org.jbehave.core.steps.StepCreator$UponFailureStep.doNotPerform(StepCreator.java:781)
	at org.jbehave.core.embedder.PerformableTree$SomethingHappened.run(PerformableTree.java:383)
	at org.jbehave.core.embedder.PerformableTree$PerformableSteps.perform(PerformableTree.java:1237)
	at org.jbehave.core.embedder.PerformableTree$AbstractPerformableScenario.perform(PerformableTree.java:1108)
	at org.jbehave.core.embedder.PerformableTree$AbstractPerformableScenario.perform(PerformableTree.java:1100)
	at org.jbehave.core.embedder.PerformableTree$NormalPerformableScenario.perform(PerformableTree.java:1134)
	at org.jbehave.core.embedder.PerformableTree$PerformableScenario.perform(PerformableTree.java:1022)
	at org.jbehave.core.embedder.PerformableTree$PerformableStory.performScenarios(PerformableTree.java:909)
	at org.jbehave.core.embedder.PerformableTree$PerformableStory.perform(PerformableTree.java:877)
	at org.jbehave.core.embedder.PerformableTree.performCancellable(PerformableTree.java:432)
	at org.jbehave.core.embedder.PerformableTree.perform(PerformableTree.java:399)
	at org.jbehave.core.embedder.StoryManager$EnqueuedStory.call(StoryManager.java:293)
	at org.jbehave.core.embedder.StoryManager$EnqueuedStory.call(StoryManager.java:266)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.io.UncheckedIOException: java.nio.file.NoSuchFileException: ./target/reports
	at org.jbehavesupport.core.internal.web.WebScreenshotCreator.createScreenshot(WebScreenshotCreator.java:50)
	at org.jbehavesupport.core.internal.web.WebScreenshotListener.onApplicationEvent(WebScreenshotListener.java:21)
	at org.jbehavesupport.core.internal.web.WebScreenshotListener.onApplicationEvent(WebScreenshotListener.java:11)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:404)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:361)
	at org.jbehavesupport.core.web.WebSteps.afterFailedScenario(WebSteps.java:91)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.jbehave.core.steps.StepCreator$MethodInvoker.invoke(StepCreator.java:995)
	at org.jbehave.core.steps.StepCreator$BeforeOrAfterStep.perform(StepCreator.java:703)
	... 18 common frames omitted
Caused by: java.nio.file.NoSuchFileException: ./target/reports
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixFileSystemProvider.createDirectory(UnixFileSystemProvider.java:388)
	at java.base/java.nio.file.Files.createDirectory(Files.java:694)
	at org.jbehavesupport.core.internal.web.WebScreenshotCreator.prepareDirectory(WebScreenshotCreator.java:91)
	at org.jbehavesupport.core.internal.web.WebScreenshotCreator.takesScreenshot(WebScreenshotCreator.java:70)
	at org.jbehavesupport.core.internal.web.WebScreenshotCreator.createScreenshot(WebScreenshotCreator.java:47)
	... 32 common frames omitted```

--- a/jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/WebScreenshotCreator.java
+++ b/jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/WebScreenshotCreator.java
@@ -66,14 +66,12 @@ public class WebScreenshotCreator {
 
     private void takesScreenshot(WebScreenshotType screenShotType) throws IOException {
         log.info("Taking {} screenshot will place it in {}", screenShotType, screenshotDirectory);
-
-        prepareDirectory();
-
         File screenshot = ((TakesScreenshot) driver).getScreenshotAs(OutputType.FILE);
 
         if (screenshot == null && screenShotType != WebScreenshotType.FAILED) {
             throw new IllegalStateException("Creation of screenshot failed");
         } else if (screenshot != null) {
+            prepareDirectory();
             File destinationFile = fileNameResolver.resolveFilePath(FILE_NAME_PATTERN, screenshotDirectory, screenShotType.toString()).toFile();
             FileUtils.copyFile(screenshot, destinationFile);
 

